cmd/buildkitd/main.go

```diff
@@ -381,33 +387,48 @@ func newController(c *cli.Context, root string) (*control.Controller, error) {
 	frontends["dockerfile.v0"] = dockerfile.NewDockerfileFrontend()
 	frontends["gateway.v0"] = gateway.NewGatewayFrontend()
 
-	ce := remotecache.NewCacheExporter(remotecache.ExporterOpt{
-		SessionManager: sessionManager,
-	})
+	resolveCacheExporter := func(xctx context.Context, xtype, xref string) (remotecache.Exporter, error) {
```

Comment:
Can we have this functionality exported from some package (registry specific exporter). Current public API would be very complicated to use with a registry exporter/importer.

Reply:
moved to cache/remotecache/registry